networkx/algorithms/approximation/tests/test_steinertree.py

```diff
@@ -208,3 +208,18 @@ def test_steiner_tree_multigraph_weight_attribute(method):
     H = nx.approximation.steiner_tree(G, list(G), method=method, weight="distance")
     assert len(H.edges) == 2 and H.has_edge(2, 0, key=1)
     assert sum(dist for *_, dist in H.edges(data="distance")) == 15
+
+
+def test_steiner_tree_method_None():
+    G = nx.star_graph(4)
+    a = nx.approximation.steiner_tree(G, [1, 3], method=None)
+    b = nx.approximation.steiner_tree(G, [1, 3], method="mehlhorn")
+    assert list(a) == list(b)
```

Comment:
This ones tricky because I'm not sure how well it actually probes the default method behavior, since both `kou` and `mehlhorn` methods should give the same result. I understand the motivation is to improve coverage, though off the top of my head I don't know of a way to test this directly.

In any case, it may be better to simply probe all three methods with a simple output directly, e.g.

```suggestion
@pytest.mark.parametrize("method", (None, "mehlhorn", "kou"))
def test_steiner_tree_methods(method):
    G = nx.star_graph(4)
    expected = nx.Graph([(0, 1), (0, 3)])
    st = nx.approximation.steinter_tree(G, [1, 3], method=method)
    assert nx.utils.edges_equal(H.edges, expected.edges)
```

Reply:
> This ones tricky because I'm not sure how well it actually probes the default method behavior, since both `kou` and `mehlhorn` methods should give the same result. I understand the motivation is to improve coverage, though off the top of my head I don't know of a way to test this directly.
> 
> In any case, it may be better to simply probe all three methods with a simple output directly, e.g.

This doesn't probe the default behavior tbh, I couldn't come up with a graph that would produce different results, but still opened up the PR hoping that with some help I'll be able to write it down.